frontend/src/toolbar/actions/EditAction.js

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect } from 'react'
+import { useActions, useValues } from 'kea'
+import { Button, Form, Input } from 'antd'
+import { actionsTabLogic } from '~/toolbar/actions/actionsTabLogic'
+import { StepField } from '~/toolbar/actions/StepField'
+import {
+    MinusCircleOutlined,
+    SearchOutlined,
+    PlusCircleOutlined,
+    CloseOutlined,
+    DeleteOutlined,
+} from '@ant-design/icons'
+
+export function EditAction() {
+    const [form] = Form.useForm()
+
+    const { initialValuesForForm, selectedActionId, inspectingElement, editingFields } = useValues(actionsTabLogic)
+    const {
+        selectAction,
+        inspectForElementWithIndex,
+        setEditingFields,
+        setForm,
+        saveAction,
+        deleteAction,
+    } = useActions(actionsTabLogic)
+
+    const { getFieldValue } = form
+
+    useEffect(() => {
+        // This sucks. We're storing the antd "form" object in kea in a reducer. Dispatching an action for it.
+        // That's so that the logic would be able to access the latest state of the form.
+        // There's another ugly hack with a `counter` selector in the actionsTabLogic as well, check it out :P
+        //
+        // I tried just saving the form's state in kea via `fields` && `onFieldsChange`, but that's in a funny
+        // format and doesn't update if the form is updated dynamically (`form.setFields(fields)` on inspect element).
+        //
+        // The solution is probably to control the form state better in the logic, for example by providing a
+        // default `fields` value (it's a bit of work) and making changes against that, not through `form.setFields`.
+        //
+        // Thanks for reading, the next coffee is on me! / Marius
+        setForm(form)
+    }, [form])
+
+    return (
+        <div>
+            <Button type="outline" size="small" onClick={() => selectAction(null)} style={{ float: 'right' }}>
+                Cancel <CloseOutlined />
+            </Button>
+            <h1 className="section-title" style={{ paddingTop: 4 }}>
+                {selectedActionId === 'new' ? 'New Action' : 'Edit Action'}
+            </h1>
+
+            <Form
+                name="action_step"
+                form={form}
+                initialValues={initialValuesForForm}
+                onFinish={saveAction}
+                fields={editingFields}
+                onFieldsChange={(changedFields, allFields) => {
+                    setEditingFields(allFields)
+                }}
+            >
+                <p>What did your user do?</p>
+                <Form.Item name="name" className="action-title-field">
+                    <Input placeholder="E.g: Clicked Sign Up" />
```

Comment:
This field should be required, otherwise it creates actions you can't click into

Reply:
Fixed! 
![image](https://user-images.githubusercontent.com/53387/85552729-977ec680-b623-11ea-8024-311bd6dc5950.png)

There are many other things I'd like to improve in the actions form, including better handling of backend errors. I added these points as separate items in this [long comment](https://github.com/PostHog/posthog/issues/870#issuecomment-644736861).